### PR TITLE
Ak/enhancing search performance

### PR DESCRIPTION
**Pull request by Akshith-Patil**
**Improving Search Functionality & Refactoring Code**

What's New:
1. Replaced search with AutoComplete component from MUI library
i. It shows a list of options as typing an input
ii. Option to clear the input field
2. Added debounce function to delay invoking `setSearchTerm` function, to avoid lag
3. Refactored code for displaying rows in the table with `useState` hook & search filtering method

Typing in the Search Box
Typing an Input

Selecting a suggestion and option to clear the search box
Clearing Search Bar

### CODE REVIEW

**Comment by mt-empty** [comment]
I'm getting this error 
```
error - Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
    at resolveDispatcher (webpack-internal:///./node_modules/react/cjs/react.development.js:1476:13)
    at useState (webpack-internal:///./node_modules/react/cjs/react.development.js:1507:20)
    at PageLoader (webpack-internal:///./components/PageLoader.tsx:96:54)
    at Profile (webpack-internal:///./pages/index.tsx:8:76)
    at processChild (evergreen/node_modules/react-dom/cjs/react-dom-server.node.development.js:3353:14)
    at resolve (evergreen/node_modules/react-dom/cjs/react-dom-server.node.development.js:3270:5)
    at ReactDOMServerRenderer.render (evergreen/node_modules/react-dom/cjs/react-dom-server.node.development.js:3753:22)
    at ReactDOMServerRenderer.read (evergreen/node_modules/react-dom/cjs/react-dom-server.node.development.js:3690:29)
    at Object.renderToString (evergreen/node_modules/react-dom/cjs/react-dom-server.node.development.js:4298:27)
    at renderPage (evergreen/node_modules/next/dist/server/render.js:684:46) {
  page: '/'
}
```

**Comment by alexdelicato** [comment]
> 

Weird, I didn't have this issue when I was testing it; was there any particular action that caused this error?

File: config.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-	"targetOrganisation": "ahmdigital",
+	"targetOrganisation": "octokit",
```
Comment (alexdelicato) [inline config.json:2]: are we keeping this as ahmdigital or octokit?

**Review by alexdelicato — APPROVED** [review]
Nice work AK, looks really good! Just remember not to merge until Thursday :)

File: config.json (hunk 1 above)
Comment (Akshith-Patil) [reply config.json:2]: Can keep AHM, but i was using octokit for testing, ecause there's more rows in the table.

**Comment by Akshith-Patil** [comment]
> I'm getting this error
> 
> ```
> error - Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
> 1. You might have mismatching versions of React and the renderer (such as React DOM)
> 2. You might be breaking the Rules of Hooks
> 3. You might have more than one copy of React in the same app
> See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
>     at resolveDispatcher (webpack-internal:///./node_modules/react/cjs/react.development.js:1476:13)
>     at useState (webpack-internal:///./node_modules/react/cjs/react.development.js:1507:20)
>     at PageLoader (webpack-internal:///./components/PageLoader.tsx:96:54)
>     at Profile (webpack-internal:///./pages/index.tsx:8:76)
>     at processChild (evergreen/node_modules/react-dom/cjs/react-dom-server.node.development.js:3353:14)
>     at resolve (evergreen/node_modules/react-dom/cjs/react-dom-server.node.development.js:3270:5)
>     at ReactDOMServerRenderer.render (evergreen/node_modules/react-dom/cjs/react-dom-server.node.development.js:3753:22)
>     at ReactDOMServerRenderer.read (evergreen/node_modules/react-dom/cjs/react-dom-server.node.development.js:3690:29)
>     at Object.renderToString (evergreen/node_modules/react-dom/cjs/react-dom-server.node.development.js:4298:27)
>     at renderPage (evergreen/node_modules/next/dist/server/render.js:684:46) {
>   page: '/'
> }
> ```

Hey Matti, how did u get the error? is this after running the dev or production build?

**Comment by mt-empty** [comment]
LGTM

**Review by mt-empty — APPROVED** [review]
LGTM

**Pull request merged**